### PR TITLE
[FEATURE] : Footer component

### PR DESCRIPTION
**Pull request by linyers**
New feature for issue https://github.com/yashchaudhari008/minime/issues/7

Contain:

- Footer component
- Footer scss module designs
- Github icon component
- Github repository page link
- yashchaudhari008 github profile link
- Contributors page link

2024-01-27-104804_649x62_scrot

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,15 @@
 import TopBar from "./components/TopBar/TopBar";
 import WidgetHolder from "./components/WidgetHolder/WidgetHolder";
 import styles from "./app.module.scss";
+import Footer from "./components/Footer/Footer";
```
Comment (yashchaudhari008) [inline src/App.tsx:4]: Move this import above styles

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import styles from "./footer.module.scss";
+import GithubIcon from "./GithubIcon";
+
+const Footer = () => {
+	return (
+		<footer className={styles.footer}>
+			<p>
+				Made with ♥ by{" "}
```
Comment (yashchaudhari008) [inline src/components/Footer/Footer.tsx:8]:
```suggestion
				Made with ♥ by
```

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+				<a target="_blank" href="https://github.com/yashchaudhari008">
+					yashchaudhari008
+				</a>{" "}
+				&{" "}
```
Comment (yashchaudhari008) [inline src/components/Footer/Footer.tsx:12]:
```suggestion
				</a>
				&
```

File: src/components/Footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+				>
+					contributors
+				</a>
+				<span className={styles.stick}> | </span>
+				<a
+					target="_blank"
+					className={styles.github}
+					href="https://github.com/yashchaudhari008/minime"
+				>
+					<GithubIcon /> Github Page
```
Comment (yashchaudhari008) [inline src/components/Footer/Footer.tsx:25]: Use `<FontAwesomeIcon icon={faGithub} className={styles.icon} />` . You might need to add `@fortawesome/free-brands-svg-icons` to add `faGithub` icon

References:
- https://fontawesome.com/docs/web/use-with/react/add-icons#add-individual-icons-explicitly
- You can also look into `src/components/widgets/AddNewWidgetBtn/AddNewWidgetBtn.tsx` to get idea to implement

File: src/components/Footer/GithubIcon.tsx
Comment (yashchaudhari008) [inline src/components/Footer/GithubIcon.tsx:1]: THIS FILE SHOULD BE DELETED as we are using **Font-Awesome** for icons

File: src/components/Footer/footer.module.scss
Comment (yashchaudhari008) [inline src/components/Footer/footer.module.scss:1]: Also try to use `px` instead of `rem`. As we are using px as standard for this project

**Review by yashchaudhari008 — CHANGES_REQUESTED** [review]
Also current alignment looks little off:

| Current | Expected |
| --- | --- |
| image | image |

**Comment by yashchaudhari008** [comment]
Also please mention yourself on the issue, so that I can assign it to you

**Comment by linyers** [comment]
While making the changes, I encountered a bug with the font-weight of the Inter font. I was able to resolve it, but I'm unsure whether to include it in this pull request or create a new one.

**Comment by yashchaudhari008** [comment]
@linyers What's the exact issue?

**Comment by linyers** [comment]
The issue was with the Inter font. When attempting to increase the weight of the Inter font, it remained at the default (400). I also noticed that certain components were using a different weight than the default, and the weight stayed at 400 even when set to 800. So, I imported various weights for the Inter font, and the issue was resolved. 

in `src/index.scss`
`@import url("https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600;700;800;900...");`

**Comment by yashchaudhari008** [comment]
Can you check whether this happens only when Chrome is opened in incognito mode? Because this is happening in my case!

**Comment by yashchaudhari008** [comment]
Also, can you push your recent changes and request re-review?

**Comment by linyers** [comment]
> Can you check whether this happens only when Chrome is opened in incognito mode? Because this is happening in my case!

I have been noticing that this seems to be some issue with browsers. I tested it on Chrome, Brave, and Firefox. I'm sharing some links that discuss this:

https://stackoverflow.com/questions/22001035/font-weight-ignored-in-chrome
https://github.com/tailwindlabs/tailwindcss/issues/1402

I have tried different solutions, and the one that worked for me was importing the weights I am going to use: 

```
@import url("http://fonts.googleapis.com/css?family=Open+Sans:400,300,500,800");
```

**Review by yashchaudhari008 — APPROVED** [review]
Looks Good! Thanks for contributing

**Pull request merged**